subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/impl/DirectorySnapshotter.java

```diff
@@ -210,7 +213,7 @@ public FileVisitResult postVisitDirectory(Path dir, @Nullable IOException exc) {
             // This way, we include each file only once.
             if (isNotFileSystemLoopException(exc)) {
                 LOGGER.debug("Could not read directory path '{}'.", dir, exc);
-                addMissingSnapshot(dir);
+                getMissingSnapshot(dir);
```

Comment:
I don't understand what's happening here. We used to create and add the thing to the builder. Now you just create it and then ignore the returned value. How does this keep working?

Also, a method called `getSomethingSomething()` where the return value is ignored is a pretty strong code smell.

Reply:
This was a leftover, thanks for noticing.
We didn't snapshot directories before, but we should model a missing directory as.
Fixed, thanks.

Reply:
I've redone this properly now.